kalite/topic_tools/__init__.py

```diff
@@ -45,24 +41,26 @@
 
 
 # Globals that can be filled
-TOPICS          = None
+TOPICS          = {}
 CACHE_VARS.append("TOPICS")
-def get_topic_tree(force=False, props=None):
-    global TOPICS, TOPICS_FILEPATH
-    if TOPICS is None or force:
-        TOPICS = softload_json(TOPICS_FILEPATH, logger=logging.debug, raises=False)
-        validate_ancestor_ids(TOPICS)  # make sure ancestor_ids are set properly
-
-        # Limit the memory footprint by unloading particular values
-        if props:
-            node_cache = get_node_cache()
-            for kind, list_by_kind in node_cache.iteritems():
-                for node_list in list_by_kind.values():
-                    for node in node_list:
-                        for att in node.keys():
-                            if att not in props:
-                                del node[att]
-    return TOPICS
+def get_topic_tree(force=False, annotate=False, channel=settings.CHANNEL):
+    global TOPICS, TOPICS_FILEPATHS
+    if TOPICS.get(channel) is None or force:
+        TOPICS[channel] = softload_json(TOPICS_FILEPATHS.get(channel), logger=logging.debug, raises=False)
+        validate_ancestor_ids(TOPICS[channel])  # make sure ancestor_ids are set properly
+
+    if annotate:
```

Comment:
Loop through all the nodes in the topic tree and cross reference with the content_cache to check availability.